### PR TITLE
Update polish translation

### PR DESCRIPTION
**Pull request by Koziolek**
Update Polish translation by:

* add B-N form grammar section
* add v1.2.3 and RegEx questions translation in FAQ
* correct grammar and words
    * Word "deprecating" is false friend in direct translation to Polish.

Small improvements in `.gitignore` by adding files generated by Jekyll, IntelliJ and asdf

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,8 @@ _site
 *.swp
 .DS_Store
 Gemfile.lock
-
+.jekyll-metadata
+.tool-versions
+semver.org.iml
```
Comment (JohnTitor) [inline .gitignore]: Please drop them, files/dirs specific to your env should go to your global gitignore config.

**Comment by JohnTitor** [comment]
@m-aciek It seems you updated the Polish translation in the past, could you review this?

File: .gitignore (hunk 1 above)
Comment (Koziolek) [reply .gitignore]: Fixed and cleaned up. 
Sorry for the inconvenience :(

File: lang/pl/index.md [hunk 2]
```diff
@@ -64,80 +64,146 @@ i „OPCJONALNY” („OPTIONAL”) w tym dokumencie należy interpretować jak
...
 7. Wersja minor Y (x.Y.z | x > 0) MUSI zostać zwiększona, jeśli nowa,
-kompatybilna wstecz funkcjonalność zostaje wprowadzona do publicznego API. MUSI
-zostać zwiększona, jeśli jakakolwiek funkcjonalność publicznego API zostaje
-zdezaprobowana. MOŻE zostać zwiększona, jeśli wprowadzone zostają nowe znaczące
-funkcjonalności lub ulepszenia w obrębie prywatnego kodu. MOŻE ona zawierać
-zmiany na poziomie patch. Numer wersji patch MUSI być ustawiony na 0, gdy wersja
-minor jest zwiększana.
+   kompatybilna wstecz funkcjonalność zostaje wprowadzona do publicznego API. MUSI
+   zostać zwiększona, jeśli jakakolwiek funkcjonalność publicznego API zostaje
+   wycofane. MOŻE zostać zwiększona, jeśli wprowadzone zostają nowe znaczące
```
Comment (m-aciek) [inline lang/pl/index.md]: "Wycofana" means withdrawn, which isn't the same as deprecated. Deprecate means "expressing a disapproval".

"[funkcjonalność] zostaje oznaczona jako wycofywana" would be a synonim here, but I'm not quite sure if it's better then current "zostaje zdezaprobowana".

File: lang/pl/index.md [hunk 3]
```diff
@@ -244,22 +310,49 @@ w semantycznym wersjonowaniu chodzi przede wszystkim o przekazanie znaczenia
...
+Nie, „v1.2.3” nie jest zgodne z wersjonowaniem semantycznym. Jednak przedrostek „v” jest
+zazwyczaj używany w języku angielskim do oznaczenia, że mamy do czynienia z numerem wersji.
+Skrót „v” od angielskiego słowa „version” jest często spotykany w systemach kontroli wersji.
+Przykładowo, w przypadku `git tag v1.2.3 -m "Release version 1.2.3"`, „v1.2.3” jest nazwą taga,
+a semantyczna wersja to „1.2.3”.
+
+### Czy istnieje sugerowane wyrażenie regularne (RegEx) do weryfikacji porawności SemVer?
+
+Istnieją dwa. Pierwsze wykorzystuje grupy nazwane i jest przeznaczone dla systemów, które
+wspierają tę funkcjonalność (PCRE [Perl Compatible Regular Expressions, np. Perl, PHP i R], Python czy Go).
```
Comment (m-aciek) [inline lang/pl/index.md:340]: (nit)
```suggestion
wspierają tę funkcjonalność (PCRE [Perl Compatible Regular Expressions, np. Perl, PHP i R], Python i Go).
```

File: lang/pl/index.md [hunk 4]
```diff
@@ -244,22 +310,49 @@ w semantycznym wersjonowaniu chodzi przede wszystkim o przekazanie znaczenia
...
+
+Patrz: https://regex101.com/r/Ly7O1x/3/
+```
+^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
+```
+
+Druga wykorzystuje numerowane grupy przechwytujące (wzór gp1 = major, gp2 = minor, gp3 = patch, gp4 = przedpremierowa i gp5 = meta-dane buildu)
+i jest kompatybilna z ECMA Script (JavaScript), PCRE [Perl Compatible Regular Expressions, np. Perl, PHP i R], Python czy Go.
```
Comment (m-aciek) [inline lang/pl/index.md]: (nit)
```suggestion
i jest kompatybilna z ECMA Script (JavaScript), PCRE [Perl Compatible Regular Expressions, np. Perl, PHP i R], Python i Go.
```

File: lang/pl/index.md [hunk 5]
```diff
@@ -244,22 +310,49 @@ w semantycznym wersjonowaniu chodzi przede wszystkim o przekazanie znaczenia
...
-### Jak powinienem radzić sobie z dezaprobowaniem funkcjonalności?
+### Jak powinienem radzić sobie z wycofywaniem funkcjonalności?
 
-Dezaprobowanie istniejącej funkcjonalności jest normalną częścią programowania
+Wycofywanie istniejącej funkcjonalności jest normalną częścią programowania
 i często jest konieczne, by móc rozwijać oprogramowanie. Gdy wycofujesz część
 swojego publicznego API, powinieneś zrobić dwie rzeczy: (1) zaktualizować
 dokumentację, by użytkownicy wiedzieli o tej zmianie, (2) wypuścić nowe wydanie
-minor z informacją o zdezaprobowaniu. Zanim całkowicie usuniesz funkcjonalność
+minor z informacją o wycofaniu. Zanim całkowicie usuniesz funkcjonalność
```
Comment (m-aciek) [inline lang/pl/index.md]:
```suggestion
minor z informacją o wycofywaniu. Zanim całkowicie usuniesz funkcjonalność
```
If we decide to change, I'd suggest rather "wycofywanie" here. I'd use not-finite form of the noun, as this is a process of withdrawing a functionality, not a one-off action.

**Review by m-aciek — CHANGES_REQUESTED** [review]
Thank you for the updates in the Polish translation.

I left a few comments. I'm not opposed to switching from "dezaprobowanie" to "wycofywanie", but currently some uses of the new word are logically incorrect. I marked them with my comments.

File: lang/pl/index.md (hunk 3 above)
Comment (Koziolek) [reply lang/pl/index.md:340]: The linguistic problem here is that when we us `i` then we create a closed set of languages. In original text 

> (PCRE [Perl Compatible Regular Expressions, i.e. Perl, PHP and R], Python and Go)

we have `and`, that could be translated as `i` (and the most of a people do that this way), but context is that we would like to present a subset of languages that support mentioned functionality. So we can use word `lub` but when we make translation PL»EN then we change original text. However, when we use `czy` then translation PL»EN will give us original text. 
In this context, Polish `czy` is not English `whether` (direct translation). 
In this [article](https://sjp.pwn.pl/poradnia/haslo/czy-jako-spojnik;14088.html) you have mentioned that `czy` could be used as equivalent of `i` or `lub`, depend on context.

File: lang/pl/index.md (hunk 5 above)
Comment (Koziolek) [reply lang/pl/index.md]: Yes. Sounds better.

File: lang/pl/index.md (hunk 2 above)
Comment (Koziolek) [reply lang/pl/index.md]: A word `zdezaprobowana` does not exist. 
In theory, the best matching word is `usunięta`, but that will be problematic with `How should I handle deprecating functionality?` section.

**Comment by JohnTitor** [comment]
@m-aciek Seems the author addressed your comments, could you re-review?

**Comment by m-aciek** [comment]
@Koziolek could you move the "deprecation" translations changes to a separate pull request? This way we won't be blocking your other contributions. "deprecation" translation still needs discussion to avoid misleading the reader.

**Review by kingofDaniel — APPROVED** [review]

**Comment by Koziolek** [comment]
@m-aciek done

File: lang/pl/index.md [hunk 6]
```diff
@@ -252,14 +318,41 @@ swojego publicznego API, powinieneś zrobić dwie rzeczy: (1) zaktualizować
 dokumentację, by użytkownicy wiedzieli o tej zmianie, (2) wypuścić nowe wydanie
 minor z informacją o zdezaprobowaniu. Zanim całkowicie usuniesz funkcjonalność
 w nowym wydaniu major, powinno być co najmniej jedno wydanie minor zawierające
-zdezaprobowanie, aby użytkownicy mogli płynnie przejść na nowe API.
+informację o zdezaprobowanie, aby użytkownicy mogli płynnie przejść na nowe API.
```
Comment (m-aciek) [inline lang/pl/index.md]:
```suggestion
informację o zdezaprobowaniu, aby użytkownicy mogli płynnie przejść na nowe API.
```

File: lang/pl/index.md [hunk 7]
```diff
@@ -252,14 +318,41 @@ swojego publicznego API, powinieneś zrobić dwie rzeczy: (1) zaktualizować
...
+Istnieją dwa. Pierwsze wykorzystuje grupy nazwane i jest przeznaczone dla systemów, które
+wspierają tę funkcjonalność (PCRE [Perl Compatible Regular Expressions, np. Perl, PHP i R], Python czy Go).
+
+Patrz: https://regex101.com/r/Ly7O1x/3/
+```
+^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
+```
+
+Druga wykorzystuje numerowane grupy przechwytujące (wzór gp1 = major, gp2 = minor, gp3 = patch, gp4 = przedpremierowa i gp5 = meta-dane buildu)
+i jest kompatybilna z ECMA Script (JavaScript), PCRE [Perl Compatible Regular Expressions, np. Perl, PHP i R], Python czy Go.
```
Comment (m-aciek) [inline lang/pl/index.md]: (nit) I think we should break the lines earlier (make the lines shorter), like in other places in the document.

**Review by m-aciek — CHANGES_REQUESTED** [review]
Spelling error and nitty line length comment. Other than that looks good to me 👍

File: lang/pl/index.md (hunk 7 above)
Comment (Koziolek) [reply lang/pl/index.md]: Fixed

File: lang/pl/index.md (hunk 6 above)
Comment (Koziolek) [reply lang/pl/index.md]: Done

**Review by m-aciek — APPROVED** [review]
LGTM, thanks @Koziolek!

**Review by JohnTitor — APPROVED** [review]
Thanks both for translating/reviewing!

**Pull request merged**